pkg/spanconfig/spanconfigjob/job.go

```diff
@@ -40,35 +39,34 @@ var _ jobs.Resumer = (*resumer)(nil)
 // Resume implements the jobs.Resumer interface.
 func (r *resumer) Resume(ctx context.Context, execCtxI interface{}) error {
 	execCtx := execCtxI.(sql.JobExecContext)
+	rc := execCtx.ConfigReconciliationJobDeps()
 
-	// TODO(zcfgs-pod): Listen in on rangefeeds over system.{descriptor,zones}
-	// and construct the right update, instead of this placeholder write over
-	// the same keyspan over and over again.
-	// TODO(zcfgs-pod): How do we test this?
-	nameSpaceTableStart := execCtx.ExecCfg().Codec.TablePrefix(keys.NamespaceTableID)
-	nameSpaceTableSpan := roachpb.Span{
-		Key:    nameSpaceTableStart,
-		EndKey: nameSpaceTableStart.PrefixEnd(),
+	updates, err := rc.Watch(ctx)
+	if err != nil {
+		log.Errorf(ctx, "error initializing sql watcher: %v", err)
 	}
 
 	for {
-		var update []roachpb.SpanConfigEntry
-		var delete []roachpb.Span
-		update = append(update, roachpb.SpanConfigEntry{
-			Span:   nameSpaceTableSpan,
-			Config: roachpb.SpanConfig{},
-		})
-
-		// NB: We cannot return from this method, as that would put the job in
-		// an un-revertable, non-running state.
-		rc := execCtx.ConfigReconciliationJobDeps()
-		if err := rc.UpdateSpanConfigEntries(ctx, update, delete); err != nil {
-			log.Errorf(ctx, "config reconciliation error: %v", err)
-		}
-
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
+		case update := <-updates:
```

Comment:
If we're not planning to add basic batching of updates here yet, leave a TODO(zcfgs-pod).